source/nodejs/adaptivecards-designer/src/card-designer.ts

```diff
@@ -2117,6 +2117,21 @@ export class CardDesigner {
         }
     }
 
+    public setSelectedPeerById(id: string) {
```

Comment:
This isn't necessary once the tree view is implemented based on the peer structure. When a tree view item is clicked, you just set its associated peer's isSelected property to true and you are done. I also now understand why you forced an Id to be generated for all actions and elements. Please do rework the tree to be based on the existing peer structure.